go/src/koding/kites/keygen/s3.go

```diff
@@ -121,22 +121,11 @@ func (ub *UserBucket) userPut(path string, rs io.ReadSeeker) (*url.URL, error) {
 		return nil, err
 	}
 
-	u := &url.URL{
+	return &url.URL{
 		Scheme: "https",
 		Path:   "/" + path,
-	}
-
-	// S3 bucket endpoints, for more details see:
-	//
-	//   http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingBucket.html
-	//
-	if ub.cfg.Region == "us-east-1" {
-		u.Host = ub.cfg.Bucket + ".s3.amazonaws.com"
-	} else {
-		u.Host = ub.cfg.Bucket + ".s3-" + ub.cfg.Region + ".amazonaws.com"
-	}
-
-	return u, nil
+		Host:   ub.cfg.Bucket + ".s3.amazonaws.com",
```

Comment:
`s3.amazonaws.com/bucket` won't work if the region is different than the default one; however `bucket.s3.amazoaws.com` will work.